docs/general/dql/selects.rst

```diff
@@ -740,6 +740,41 @@ an upper-bound::
     Both the ``from`` and ``to`` index values are inclusive.
     Using an index greater than the array size results in an empty array.
 
+For selecting records with values of empty or unassigned arrays, like::
+
+    cr> insert into locations (id, name, position, kind, landmarks) values
+    ... (15, 'Voidspace', 0, 'Milliways', []);
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+you can use the :ref:`array_length <scalar-array_length>` scalar function::
+
+    cr> select id, landmarks from locations where
+    ... array_length(landmarks, 1) = 0 order by id;
+    +-----+-----------+
+    |  id | landmarks |
+    +-----+-----------+
+    | 15  | []        |
+    +-----+-----------+
+    SELECT 1 row in set (... sec)
+
+It is common to match for both ``NULL`` and empty values, like::
+
+    cr> select count(*) from locations where
+    ... landmarks is null or
+    ... array_length(landmarks, 1) = 0;
```

Comment:
Following up on @mfussenegger's response at https://github.com/crate/crate/issues/9398#issuecomment-1169750540: Would that also be a valid query to match empty or unassigned values in `array`-type columns, without needing the `array_length` function at all? Is it inefficient?

```sql
select * from locations where landmarks is null or landmarks = [];
```

Reply:
Adjusted with 99242018.